src/utils/recommendations/preprocessing/lemmatizer.js

```diff
@@ -0,0 +1,7 @@
+const lemmatizer = require('wink-lemmatizer');
+
+function lemmatize(tokens) {
+  return tokens.map((token) => lemmatizer.noun(token) || lemmatizer.verb(token) || lemmatizer.adjective(token) || token);
```

Comment:
Can we try to lemmatize in place instead of creating a new array & returning it?